app/jobs/create_member_email_batch_job.rb

```diff
@@ -14,52 +14,37 @@ def perform
     current_community = Community.where(id: community_id).first
 
     Delayed::Job.transaction do
-      recipient_ids(mode, current_community).each do |recipient_id|
-        Delayed::Job.enqueue(CommunityMemberEmailSentJob.new(sender_id, recipient_id, community_id, content, locale))
+      community_members(mode, current_community).find_each do |recipient|
```

Comment:
All you need from each person is the ID here, so it would be more efficient to pluck it and loop over the IDs. That way you avoid transferring tons of unnecessary data from DB.

Something like:

```
community_members(mode, current_community).pluck(:id).each do |recipient_id|
   ...
end
```

Reply:
Could not agree Your point of view. That pluck is efficient for 100 members. find_each covers any record count 1 or 1000000 without CPU/RAM overload. Probably ActiveRecord scope can be combined with select scope.select(something) (not sure right now).

Reply:
Good point, @aivils. Since pluck would return array of IDs, that's probably not too much memory even there are hundreds of thousands of IDs. An ID is 22 bytes + some overhead, so e.g. 500 000 IDs would be little more than 10 MB in size.

On the other hand find_each is more efficient on the app code side, but would require the complex queries for finding the users to be executed multiple times (for each batch), resulting possibly to a bit higher load on the DB.

I'm not sure which is better, without testing more carefully, so I leave it to you. Either way is probably ok, if we make the queries efficient enough.